internal/fetcher/fetcher.go

```diff
@@ -153,7 +165,7 @@ func (f *fetcher) filterFormats(formats map[file.Format]driver.Share) map[file.F
 	fs := make(map[file.Format]driver.Share)
 	for format, share := range formats {
 		for _, vf := range f.Formats {
-			if format == vf {
+			if format == vf && matchKeywords(share.FileName, f.Keywords) {
```

Comment:
I don't think we should add the name filter here. Instead, I think we may need to add a new filter instead.
BTW, we should not mark the book downloaded if we just filter it instead download it. This may require more designs here.